### PR TITLE
Add and run `pre-commit`

### PR DESCRIPTION
**Pull request by paddyroddy**

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[flake8]
+max-line-length = 88
```
Comment (paddyroddy) [inline .flake8:2]: to please `black`

File: napari_btrack/_dock_widget.py [hunk 2]
```diff
@@ -31,7 +32,7 @@ def _on_click(self):
 
 
 @magic_factory
-def example_magic_widget(img_layer: "napari.layers.Image"):
+def example_magic_widget(img_layer: napari.layers.Image):
```
Comment (paddyroddy) [inline napari_btrack/_dock_widget.py:35]: I assume this is just usual typing and, as such, quotes shouldn't be here?

File: napari_btrack/_tests/test_dock_widget.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
-from napari_btrack import napari_experimental_provide_dock_widget
 import pytest
 
+from napari_btrack import napari_experimental_provide_dock_widget  # noqa: F401
```
Comment (paddyroddy) [inline napari_btrack/_tests/test_dock_widget.py:3]: This isn't being used currently, not sure if intentional. `# noqa: F401` tells `flake8` not to worry

**Review by dstansby — APPROVED** [review]
👍 . Do you want to leave adding pre-commit to the CI to another PR? If so feel free to self-merge.

**Pull request merged**